### PR TITLE
edge-21.3.1 change notes

### PR DESCRIPTION
**Pull request by alpeb**
## edge-21.3.1

This edge release is another release candidate, bringing us closer to
`stable-2.10.0`! It fixes the Helm install/upgrade procedure and ships some new
CLI commands, among other improvements.

* Fixed Helm install/upgrade, which was failing when not explicitly setting
  `proxy.image.version`
* Added a warning in the dashboard when viewing tap streams from resources that
  don't have tap enabled
* Added the command `linkerd viz list` to list meshed pods and indicate which can
  be tapped, which need to be restarted before they can be tapped, and which
  have tap disabled
* Similarly, added the command `linkerd jaeger list` to list meshed pods and
  indicate which will participate in tracing
* Added the `--opaque-ports` flag to `linkerd inject` to specify the list of
  opaque ports when injecting pods (and services)
* Simplified the output of `linkerd jaeger check`, combining the checks for the
  status of each component into a single check
* Changed the destination component to receive the list of default opaque ports
  set during install so that it's properly reflected during discovery
* Moved the level of the proxy server's I/O-related "Connection closed" messages
  from info to debug, which were not providing actionable information

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+* Similarly, added the command `linkerd jaeger list` to list meshed pods and
+  indicate which will participate in tracing
+* Added the `--opaque-ports` flag to `linkerd inject` to specify the list of
+  opaque ports when injection pods (and services)
+* Simplified the output of `linkerd jaeger check`, combining the checks for the
+  status of each component into a single check
+* Had the destination component receive the `opaquePorts` config set during
+  install so it's properly reflected during injection and discovery
+* Moved the level of the proxy server's I/O errors from info to debug, which
+  were not providing actionable information
```
Comment (olix0r) [inline CHANGES.md:23]:
```suggestion
* Moved the level of the proxy server's I/O-related "Connection closed" messages
  from info to debug, which were not providing actionable information
```

**Review by olix0r — APPROVED** [review]

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
 
+## edge-21.3.1
+
+This edge release is another release candidate, bringing us closer to
+`stable-2.10.0`! It fixes the Helm install/upgrade procedure and ships some new
+CLI commands, among other improvements.
+
+* Fixed Helm install/upgrade, which was failing when not explicitly setting
+  `proxy.image.version`
+* Added the command `linkerd viz list` to list mesh pods and indicate which can
```
Comment (kleimkuhler) [inline CHANGES.md:11]:
```suggestion
* Added the command `linkerd viz list` to list meshed pods and indicate which can
```

File: CHANGES.md [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+  be tapped, which need to be restarted before they can be tapped, and which
+  have tap disabled
+* Similarly, added the command `linkerd jaeger list` to list meshed pods and
+  indicate which will participate in tracing
+* Added the `--opaque-ports` flag to `linkerd inject` to specify the list of
+  opaque ports when injection pods (and services)
```
Comment (kleimkuhler) [inline CHANGES.md:17]:
```suggestion
  opaque ports when injecting pods (and services)
```

File: CHANGES.md [hunk 4]
```diff
@@ -1,5 +1,27 @@
...
+* Simplified the output of `linkerd jaeger check`, combining the checks for the
+  status of each component into a single check
+* Had the destination component receive the `opaquePorts` config set during
+  install so it's properly reflected during injection and discovery
```
Comment (kleimkuhler) [inline CHANGES.md:21]: Maybe reword this a little?

```suggestion
* Changed the destination component to receive the list of default opaque ports
  set during install so that it's properly reflected during discovery
```

**Review by kleimkuhler — APPROVED** [review]
